### DISCUSSION TITLE
Users who log in via Google, Facebook etc., user accounts are not created

### DISCUSSION THREAD

**Discussion by marcoriesco** [nextauthjs/next-auth#1389] [Help]
**Your question**
Users who log in via Google, Facebook etc., user accounts are not created. I have some doubts about this, because my way of thinking may be wrong.

In the ideal world, the user logs in, via Google, for example, then is redirected to another screen where he completes his registration, in which his user account has already been created.

Question, how to link the access account with a user account?

It's possible ?

**Expected behavior**
In my current scenario, I need exactly that, since the user must have the ability to create events, on my frontend. I have already tested and with login only access I am not allowed for anything.

* [ ] Found the documentation helpful
* [ ] Found documentation but was incomplete
* [x] Could not find relevant documentation
* [ ] Found the example project helpful
* [ ] Did not find the example project helpful

**Comment by Rashmi-278** [comment]
they are created when the sign up using oauth , what database are you using ?

**Comment by marcoriesco** [comment]
Thanx for response, i'm using Mongodb.

In response, two entries are created in the bank, one in ACCOUNTS and the other in USERS, in my project I need to link Events and Orders to users, but with this type of access it does not work.

**Comment by iaincollins** [comment]
> Thanx for response, i'm using Mongodb.
> 
> In response, two entries are created in the bank, one in ACCOUNTS and the other in USERS, in my project I need to link Events and Orders to users, but with this type of access it does not work.

Hi there!

So the `accounts` collection (which contains OAuth accounts linked to that User) references the User ID in the `userId` property.

If you have another collection, like `events` or `orders`, then you can do something similar and include a `userId` property or array in each event. With a NoSQL DB like Mongo you can also easily add additional properties to a User object once it has been created.

**Comment by marcoriesco** [comment]
I understand your point of view, but I believe that the relationship between user and (order, product, etc.) has to be made within the visual backend. Anyway, I was unable to link a "registered" user via social media to any collection.

**Comment by marcoriesco** [comment]
What is also very relevant, as these users are not created, it is impossible to create (posts, products), whatever for .. with an authenticated user, because the generated token is not recognized by Strapi.
